### ISSUE TITLE
Variable.ref() no longer exists as of TensorFlow v0.12

### ISSUE THREAD

**Issue by shoyer** [blei-lab/edward#367]
It's now a [private method `Variable._ref()`](https://github.com/tensorflow/tensorflow/blob/r0.12/tensorflow/python/ops/variables.py#L460))

Edward uses this in `edward/inferences/klpq.py ` and `edward/inferences/klqp.py` for computing gradients, in lines that look like:
```python
    grads = tf.gradients(
        -tf.reduce_mean(q_log_prob * tf.stop_gradient(w_norm)),
        [v.ref() for v in var_list])
```

I believe it should work if `[v.ref() for v in var_list]` is simply replaced by `var_list`.

**Comment by dustinvtran** [comment]
the proposed solution makes sense. we should check this; i forget why but i recall having to add `ref()` for a reason. maybe it was because `tf.gradients()` didn't work explicitly on tensorflow variables and do such a conversion internally? (vague recollection)

**Comment by gokceneraslan** [comment]
I addressed this and a few other TF 0.12 incompatibilities in #374.

**Comment by dustinvtran** [comment]
looking back, i used `[v.ref() for v in var_list]` instead of `var_list` because the `compute_gradients()` method in TensorFlow optimizers does it: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/training/optimizer.py#L347

this looks like overkill for us so indeed we can drop it.
